oidc-api/src/main/java/uk/gov/di/authentication/oidc/services/RequestObjectService.java

```diff
@@ -97,8 +96,9 @@ public Optional<AuthRequestError> validateRequestObject(AuthenticationRequest au
 
             var redirectURI = URI.create((String) jwtClaimsSet.getClaim("redirect_uri"));
 
-            if (!ClientType.APP.getValue().equals(client.getClientType())) {
-                LOG.error("ClientType of client is not 'app'");
+            if (!ClientType.APP.getValue().equals(client.getClientType())
+                    && !ClientType.WEB.getValue().equals(client.getClientType())) {
+                LOG.error("ClientType of client is not 'app' or 'web'");
```

Comment:
This is now all of the current client types. What's the intention here: is it to check it's specifically one of `web` or `app`, or to check it's a valid client type? What should happen when we add a new one?

Reply:
App was the distinguishing factor in the client registry and was effectively the "This RP uses request objects".

This is validating the content of the field, but it could just as easily be :

`Arrays.asList(ClientType.values()).contains(client.getClientType())`

Reply:
Yep, let's go for this to make the intention clear 👍